### ISSUE TITLE
Question: Pass result to dependent task

### ISSUE THREAD

**Issue by gamefundas** [dexecutor/dexecutor-ignite#1]
I just started reading about this project and it sounds exciting. Had built something similar couple of years back, called GraphExecutor, based on JGraphT library which had somewhat similar features but its kind of outdated now. 

One of the features I had there was the ability to pass context(result, message etc) from one task to the other. Basically dependent jobs received the result from the parent tasks so that you can form processing streams. 

Now I haven't looked at the API fully but would like to know if this is supported, as essentially I can replace what I have with this new framework in combination with Apache Ignite.

**Comment by mnadeem** [comment]
Not sure if you get a chance to look into 
https://github.com/dexecutor/dexecutor-core/wiki/How-Do-I-%3F#how-do-i-execute-tasks-conditionally-based-on-the-result-of-parent-tasks

Hopefully it serve your purpose.. Further I have enabled issue tracking in core project.

**Comment by gamefundas** [comment]
I did look at this but in order to use the parent results in the child task it would be nice to have some thing like 
protected Task<K,V> getParentResults() in TaskProvider

Don't fully understand the lifecycle of TaskProvider and when shouldExecute() gets invoked but I think if we made the results easily accessible then people will be able to use it in execute() block instead of copying references from shouldExecute()

Just a thought.

**Comment by mnadeem** [comment]
I believe, The only place, parent tasks result is required, when executing a given task, i.e, within a context of task, Lets say a task 'R' Depends on task 'P' and and 'Q', While executing Task 'R', Parent tasks (Q and R) results would be provide when the following method is overridden in `Task`

`public boolean shouldExecute(ExecutionResults<String, String> parentResults)`

Hope it makes sense!

**Comment by mnadeem** [comment]
Probably you can provide me your use case, that would help a ton

**Comment by mnadeem** [comment]
Closing this issue, since we don't have use case
